src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -43,7 +43,8 @@ public IDisposable Start(IServerInformation serverInformation, Func<IFeatureColl
             {
                 disposables.Add(engine.CreateServer(
                     address.Scheme,
-                    address.Host,
+                    // Unix sockets use a file path, not a hostname.
+                    address.Scheme == Constants.UnixScheme ? address.Path : address.Host,
```

Comment:
Yeah, looking at how `unix:` is used it appears it's treated as the "host" information rather than the "scheme". Which kind of makes sense - it determines where to connect/listen rather than how you frame data on the stream after the connection.

Also, if fastcgi is added as a way to frame data - that would likely appear to kestrel as a different scheme which would work on either tcp network sockets or unix domain sockets.

So all of the below would be valid:
* `http://127.0.0.1:8080/`
* `http://unix:/tmp/my-server/:/`
* `fastcgi://127.0.0.1:9090/`
* `fastcgi://unix:/tmp/my-other-server/:/`

So that would involve two changes... Kestrel needs to understand `unix:*:` is a valid host format, allowing everything after the second : to become the path, and then you would look for `unix:` leading characters in the host to tell that it's not a tcp connection.